File: python/python-sdk/kestrapy/api/banners_api.py
from typing import List, Optional

from kestrapy.base_api import BaseApi
from kestrapy.models.banner import Banner
from kestrapy.models.query_filter import QueryFilter


class BannersApi(BaseApi):
    """Announcement banners (`/api/v1/banners`). Instance-owner-only."""

    def create_banner(self, banner: Banner) -> Banner:
        path = self._superadmin_path("banners")
        return self._json_request("POST", path, Banner, body=banner)

    def update_banner(self, id: str, banner: Banner) -> Banner:
        path = self._superadmin_path("banners", id)
        return self._json_request("PUT", path, Banner, body=banner)

    def delete_banner(self, id: str) -> None:
        path = self._superadmin_path("banners", id)
        self._void_request("DELETE", path)

    def search_banners(self, filters: Optional[List[QueryFilter]] = None) -> List[Banner]:
        path = self._superadmin_path("banners", "search")
        params: list = []
        self._append_filter_params(params, filters)
        return self._json_list_request("GET", path, Banner, params=params)

File: go-sdk/kestra_api_client/banners_api.go
package kestra_api_client

import "context"

// BannersAPI covers the instance-level /api/v1/banners routes. Banners are
// instance-wide announcements; every endpoint requires the caller to be an
// instance owner and is therefore not tenant-scoped.
type BannersAPI struct {
	baseAPI
}

// SearchBanners returns every banner. The optional filters use the BANNER query
// filter format; pass nil to list them all.
func (a *BannersAPI) SearchBanners(ctx context.Context, filters []SearchFilter) ([]Banner, error) {
	params := buildQueryParams()
	appendFilterParams(params, filters)
	return doJSON[[]Banner](&a.baseAPI, ctx, "GET", superadminPath("banners", "search"), nil, params)
}

// CreateBanner creates a banner. The server forces it active and clears any
// tenantId in the body, so the returned banner carries the server-assigned id.
func (a *BannersAPI) CreateBanner(ctx context.Context, banner Banner) (*Banner, error) {
	result, err := doJSON[Banner](&a.baseAPI, ctx, "POST", superadminPath("banners"), banner, nil)
	if err != nil {
		return nil, err
	}
	return &result, nil
}

// UpdateBanner updates the banner with the given id. The body id must match the
// path id or the server answers 422.
func (a *BannersAPI) UpdateBanner(ctx context.Context, id string, banner Banner) (*Banner, error) {
	result, err := doJSON[Banner](&a.baseAPI, ctx, "PUT", superadminPath("banners", id), banner, nil)
	if err != nil {
		return nil, err
	}
	return &result, nil
}

// DeleteBanner deletes the banner with the given id.
func (a *BannersAPI) DeleteBanner(ctx context.Context, id string) error {
	return a.doVoid(ctx, "DELETE", superadminPath("banners", id), nil, nil)
}
